aspnetcore/host-and-deploy/proxy-load-balancer.md

```diff
@@ -61,32 +61,39 @@ Configure the middleware with <xref:Microsoft.AspNetCore.Builder.ForwardedHeader
 ```csharp
 public void ConfigureServices(IServiceCollection services)
 {
-    services.AddMvc();
-
+    services.AddControllersWithViews();
     services.Configure<ForwardedHeadersOptions>(options =>
     {
-        options.ForwardedHeaders = 
+        options.ForwardedHeaders =
             ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
     });
 }
 
-public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 {
-    app.UseForwardedHeaders();
-
     if (env.IsDevelopment())
     {
         app.UseDeveloperExceptionPage();
     }
     else
     {
         app.UseExceptionHandler("/Home/Error");
+        app.UseHsts();
     }
-
+    app.UseForwardedHeaders();
```

Comment:
The trouble with this ordering is that UseHsts depends on the request scheme being updated already.
https://github.com/dotnet/aspnetcore/blob/c95ee2b051814b787b07f55ff224d03d550aafeb/src/Middleware/HttpsPolicy/src/HstsMiddleware.cs#L72

It either needs to go above the if blocks or inside each of them after diagnostics and before Hsts.